Source/JavaScriptCore/inspector/protocol/CSS.json

```diff
@@ -479,12 +483,12 @@
             ]
         },
         {
-            "name": "nodeLayoutContextTypeChanged",
-            "description": "Called when a node's layout context type has changed.",
+            "name": "nodeLayoutFlagsChanged",
+            "description": "Called when the layout of a node changes.",
             "targetTypes": ["page"],
             "parameters": [
-                { "name": "nodeId", "$ref": "DOM.NodeId", "description": "Identifier of the node whose layout context type changed." },
-                { "name": "layoutContextType", "$ref": "LayoutContextType", "optional": true, "description": "The new layout context type of the node. When not provided, the <code>LayoutContextType</code> of the node is not a context for which Web Inspector has specific functionality." }
+                { "name": "nodeId", "$ref": "DOM.NodeId", "description": "Identifier of the node whose layout changed." },
+                { "name": "layoutFlags", "type": "array", "items": { "type": "string", "description": "CSS.LayoutFlag" }, "optional": true, "description": "Relevant information about the layout of the node. When not provided, the layout of the node is not important to Web Inspector." }
```

Comment:
`"items": { "$ref": "LayoutFlag" }`?

Reply:
sadly we can't do that because the C++ protocol generator code doesn't know how to auto-convert a `JSON::ArrayOf` with an enum into the actual `JSON::ArrayOf<String>` that's sent to the frontend

and we can't create a specialization of `JSON::ArrayOf::addItem` because that exists in WTF, which is below `Inspector::Protocol::Helpers::getEnumConstantValue` (also that `getEnumConstantValue` is overloaded by WebDriver/`Automation` stuff, so we'd potentially have naming clashes)

this is kinda similar to how objects sent to the backend show up as `JSON::Object` instead of the type in the protocol, since we don't have sophisticated enough logic to auto-convert from the raw JSON to the `Inspector::Protocol` object